### DISCUSSION TITLE
API: cannot create virtual machine due to cluster required error while filled in.

### DISCUSSION THREAD

**Discussion by dannylifino** [netbox-community/netbox#6968] [Q&A] [answered]
### NetBox version

v2.9.8

### Python version

3.9

### Steps to Reproduce

Make a POST api call to /virtualization/virtual-machines/ with data down below.

`Array
(
    [name] => vps131.servername.com
    [cluster] => 2
)
`

### Expected Behavior

The virtual machine to be created, because the required fields name and cluster are filled in.

### Observed Behavior

The cluster field is required is the error it gives. This is some weird behaviour since the name in the request is found, but the cluster is not.

Client error: `POST api/virtualization/virtual-machines/` resulted in a `400 Bad Request` response:
{"cluster":["This field is required."]}

If i remove the name field from the request, it will also show: the name field is required.

**Comment by sdktr** [comment]
Can you try and reproduce your request against demo.netbox.dev? If reproducable there, please post the curl request other people can try. Your current request is from a higher level language then the raw json that goes over the wire.

**Comment by jeremystretch** [comment, marked as answer]
> `Array ( [name] => vps131.servername.com [cluster] => 2 )`

This isn't JSON data. Please see the [REST API documentation](https://netbox.readthedocs.io/en/stable/rest-api/overview/) for examples of valid API requests.
